salt/version.py

```diff
@@ -33,6 +34,200 @@
 # <---- ATTENTION ----------------------------------------------------------------------------------------------------
 
 
+@total_ordering
+class SaltVersion(namedtuple("SaltVersion", "name, info, released")):
```

Comment:
What problems is this new way of retrieving / handling salt versioning throughout this file change? And is this also why there is modification being done in `setup.py`?

Reply:
Previously we relied on uncommenting the version names and we released. Part of it was because we did not know when they were going to be released since they were date based.
Since 3000, we now know all version numbers.
Additionally, the next release would be the last uncommented release, which wasn't really the next release(the method), commonly it would be 2 or 3 releases ahead.

And yes, because all versions are now uncommented, the change to generate the version had to be done, because we don't upload the `.git` directly to the started VMs, so there was no way of knowing the actual current release.